vignettes/concepts_conventions.Rmd

```diff
@@ -0,0 +1,285 @@
+---
+title: "Programming Concepts and Conventions"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Programming Concepts and Conventions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+
+library(dplyr)
+library(admiral)
+library(admiraldev)
+```
+
+# Introduction
+
+This vignette aims to discuss some of the common programming concepts and conventions that have been adopted within the
+`{admiral}` family of packages. It is intended to be a user-facing version of the [Programming Strategy](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html) page, but users can also read the
+latter after becoming more familiar with the package to expand further on any topics of interest. For some of common
+`{admiral}` FAQ, visit the corresponding [FAQ page](faq.html).
+
+# Input and Output
+
+It is expected that the input dataset is not grouped. Otherwise an error is
+issued.
+
+The output dataset is ungrouped. The observations are not ordered in a dedicated
+way. In particular, the order of the observations of the input dataset may not
+be preserved.
+
+# `{admiral}` Functions and Options
+
+As a general principle, the behavior of the `{admiral}` functions is only determined by their input, not by any global object,
+i.e. all inputs like datasets, variable names, options, etc. must be provided to the function by arguments. Correspondingly, the
+functions do not have any side-effects like creating or modifying global objects, printing, writing files, etc.
```

Comment:
IS this true?  I thought some of the source stuff printed stuff?

Reply:
clarified that this is "in general"